packages/datagateway-download/src/downloadConfirmation/downloadConfirmDialog.component.tsx

```diff
@@ -161,15 +212,130 @@ const DownloadConfirmDialog: React.FC<DownloadConfirmDialogProps> = (
     if (isSubmitSuccessful) clearCart();
   };
 
-  useEffect(() => {
+  // Broadcast a SciGateway notification for any error encountered.
+  const broadcastError = (errorMessage: string): void => {
+    document.dispatchEvent(
+      new CustomEvent(MicroFrontendMessageId, {
+        detail: {
+          type: 'scigateway:api:notification',
+          payload: {
+            severity: 'error',
+            message: errorMessage,
+          },
+        },
+      })
+    );
+  };
+
+  // Sort access methods into a sorted array used for
+  // rendering the select dropdown list.
+  const sortMethods = React.useCallback(() => {
+    const sorted = Object.entries(statusMethods).sort(
+      ([, methodInfoA], [, methodInfoB]) => {
+        let res =
+          (methodInfoA.disabled !== undefined
+            ? methodInfoA.disabled === false
+              ? -1
+              : 0
+            : 1) -
+          (methodInfoB.disabled !== undefined
+            ? methodInfoB.disabled === false
+              ? -1
+              : 0
+            : 1);
+        return res;
+      }
+    );
+
+    // Set the sorted, selected access method
+    // and set it to have been sorted.
+    setSortedMethods(sorted);
+    setSelectedMethod(sorted[0][0]);
+    setIsSorted(true);
+  }, [statusMethods, setSortedMethods, setSelectedMethod, setIsSorted]);
+
+  React.useEffect(() => {
+    async function getStatus(): Promise<void> {
+      let statusErrors: string[] = [];
+      Promise.all(
+        Object.entries(statusMethods).map(([method]) =>
+          getDownloadTypeStatus(method, {
+            facilityName: settings.facilityName,
+            downloadApiUrl: settings.downloadApiUrl,
+          })
+        )
+      ).then(methodStatus => {
```

Comment:
Since this is a `Promise.all`, this object is an array right? So comprises of multiple `methodStatus`. I know that status can be plural as well as singular but it would probably be clearer if it was statuses.